joomla/joomla-cms
# Dynamic Editor-Buttons TinyMCE

Pull Request for Issue # .

### Summary of Changes
- Make TinyMCE-Plugins (Buttons) configurable (enable / disable, accessrights etc..)
- Makes easier to extend TinyMCE
- easier to maintenance
![vorhernachherdragdrop](https://cloud.githubusercontent.com/assets/9518007/18065810/d20abc94-6e35-11e6-9ed8-58113cee4d26.png)

### Testing Instructions
Before:
You cant enable / disable / install / uninstall TinyMCE-Buttons with Joomla
After:
I created an example for TinyMCE-Smileys-Plugin
1. Download https://github.com/alimpam/tinymce-smileys-plugin and install it
2. In the Plugin-Manager you will see a new Plugin **Smileys-Plugin for TinyMCE** (belongs to editors_buttons category) which can be enabled / disabled
3. You can edit an article and checkout if it works. So if it is enabled you can insert smileys ;-)


### Documentation Changes Required

This PR makes it easier to expand the functionality of TinyMCE and
allows to install/uninstall (as Joomla-Plugins) or enable/disable
Buttons. For that there is a new plugin category "editors_buttons" which
must be used. Also jdragdrop is fitted for this changes. (based on #11157)
